File: lessons/pediy/01/keygen.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char *argv[]){
	if (argc != 2){
		printf("Usage : \n\t%s [username]\n", argv[0]);
		exit(1);
	}

	char data[] = {0x0C, 0x0A, 0x13, 0x09, 0x0c, 0x0B, 0x0A, 0x08};
	char *username = argv[1];
	int length = strlen(username);
	int i;
	int j;
	int password = 0;
	for (i = 3, j = 0; i < length; i++, j++){
		password += username[i] * data[j % 8];
	}

	printf("Username : [ %s ]\nPassword : [ %d ]\n", username, password);

	return 0;
}

File: lessons/pediy/01/keygen.py
#!/usr/bin/env python
# encoding:utf-8

import sys

def get_key(username):
    data = [0x0C, 0x0A, 0x13, 0x09, 0x0C, 0x0B, 0x0A, 0x08]
    key = 0
    i = 3
    j = 0
    while True:
        if i >= len(username):
            break
        key += ord(username[i]) * data[j % 8]
        i += 1
        j += 1
    return key

def show_help():
    print "Usage : \n\tpython %s [username]" % sys.argv[0]

def main():
    if len(sys.argv) != 2:
        show_help()
        exit(1)
    print get_key(sys.argv[1])

if __name__ == "__main__":
    main()
